### PR TITLE
Pre-renderpass changes

### PR DESCRIPTION
**Pull request by Kangz**
@kainino0x, @SenorBlanco PTAL, these are changes required for before the PR that will change NXT to use "Metal style" render passes instead of "Vulkan style" ones.

### CODE REVIEW

File: src/backend/metal/CommandBufferMTL.mm [hunk 1]
```diff
@@ -144,7 +144,7 @@ void BeginSubpass(id<MTLCommandBuffer> commandBuffer, uint32_t subpass) {
                 }
 
                 render = [commandBuffer renderCommandEncoderWithDescriptor:descriptor];
-                [descriptor release];
```
Comment (SenorBlanco) [inline src/backend/metal/CommandBufferMTL.mm]: As discussed, let's do these changes in a separate CL.

Comment (Kangz) [reply src/backend/metal/CommandBufferMTL.mm]: Done

**Comment by Kangz** [comment]
Thanks for the reviews, @SenorBlanco will put up a PR for the release fix.
